index.js

```diff
@@ -918,23 +918,23 @@ class Hyperdrive extends Nanoresource {
       }
     })
 
-    return unmirror
+    this._unmirror = unmirror
+    return this._unmirror
 
     function unmirror () {
-      if (!self._mirrorRanges) return
-      const ranges = self._mirrorRanges
-      self._mirrorRanges = null
+      if (!self._unmirror) return
+      self._unmirror = null
       self.removeListener('content-feed', oncore)
       self.removeListener('metadata-feed', oncore)
-      for (const [ core, range ] of ranges) {
+      for (const [ core, range ] of mirrorRanges) {
         core.undownload(range)
       }
     }
 
     function oncore (core) {
       if (!core) return
-      if (!self._mirrorRanges || self._mirrorRanges.has(core)) return
-      self._mirrorRanges.set(core, core.download({ start: 0, end: -1 }))
+      if (!self._unmirror || mirrorRanges.has(core)) return
```

Comment:
You need || self._unmirror !== unmirror here as well to avoid a race condition

Reply:
Yep added this